servo/connectors/kubernetes.py

```diff
@@ -2003,6 +2016,578 @@ async def get_restart_count(self) -> int:
+
+    async def get_status(self) -> RolloutStatus:
+        """Get the status of the Rollout.
+        Returns:
+            The status of the Rollout.
+        """
+        self.logger.info(f'checking status of rollout "{self.name}"')
+        # first, refresh the rollout state to ensure the latest status
+        await self.refresh()
+
+        # return the status from the rollout
+        return self.obj.status
+
+    async def get_pods(self) -> List[Pod]:
+        """Get the pods for the Rollout.
+
+        Returns:
+            A list of pods that belong to the rollout.
+        """
+        self.logger.debug(f'getting pods for rollout "{self.name}"')
+
+        async with Pod.preferred_client() as api_client:
+            label_selector = self.obj.spec.selector.match_labels
+            pod_list:kubernetes_asyncio.client.V1PodList = await api_client.list_namespaced_pod(
+                namespace=self.namespace, label_selector=selector_string(label_selector)
+            )
+
+        pods = [Pod(p) for p in pod_list.items]
+        return pods
+
+    @property
+    def status(self) -> RolloutStatus:
+        """Return the status of the Rollout.
+        Returns:
+            The status of the Rollout.
+        """
+        return self.obj.status
+
+    async def is_ready(self) -> bool:
+        """Check if the Rollout is in the ready state.
+
+        Returns:
+            True if in the ready state; False otherwise.
+        """
+        await self.refresh()
+
+        # if there is no status, the deployment is definitely not ready
+        status = self.obj.status
+        if status is None:
+            return False
+
+        # check the status for the number of total replicas and compare
+        # it to the number of ready replicas. if the numbers are
+        # equal, the deployment is ready; otherwise it is not ready.
+        total = status.replicas
+        ready = status.ready_replicas
+
+        if total is None:
+            return False
+
+        return total == ready
+
+    @property
+    def containers(self) -> List[Container]:
+        """
+        Return a list of Container objects from the underlying pod template spec.
+        """
+        return list(
+            map(lambda c: Container(c, None), self.obj.spec.template.spec.containers)
+        )
+
+    def find_container(self, name: str) -> Optional[Container]:
+        """
+        Return the container with the given name.
+        """
+        return next(filter(lambda c: c.name == name, self.containers), None)
+
+    async def get_tuning_container(self, config: ContainerConfiguration) -> Optional[Container]:
+        tuning_container = self.find_container(config.name)
+        if tuning_container is not None:
+            async with kubernetes_asyncio.client.ApiClient() as api_client:
+                tuning_container.obj = api_client.deserialize(
+                        response=FakeKubeResponse(tuning_container.obj.dict(by_alias=True, exclude_none=True)),
```

Comment:
Wait, what? Why are we returning a fake response? And how is this `get_tuning_container` since it has no direct tie to the concept of a tuning -- which is at the `Pod` not `Container` level?

Reply:
RE: the fake response, its also part of the workaround mentioned here: https://github.com/opsani/servox/pull/303#discussion_r685448040

I will update the method name to something more appropriate for the functionality